app/models/dob.rb

```diff
@@ -0,0 +1,74 @@
+class Dob < Base
+
+  attribute :date_of_birth, :datetime
+  
+  validates :date_of_birth, presence: true
+  
+  def next_step
+   "uk_or_overseas" 
+  end
+
+  def initialize(attributes = nil)
+    super
+
+    assign_attributes(attributes) if attributes
+  end
+
+  def assign_attributes(attributes)
+    new_attributes = attributes.stringify_keys
+    multiparameter_attributes = extract_multiparameter_attributes(new_attributes)
+
+    multiparameter_attributes.each do |multiparameter_attribute, values_hash|
+      set_values = (1..3).collect{ |position| values_hash[position].to_i }
+   
+    if set_values.include?(0)
+      errors.add(:date_of_birth, "can't be blank") 
+      return
+    elsif set_values[0] < 1920 || set_values[0] > Date.today.year
+      errors.add(:date_of_birth, "not a valid year")
+      return
+    elsif set_values[1] < 1 || set_values[1] > 12
+      errors.add(:date_of_birth, "not a valid month")
+      return
+    elsif set_values[2] < 1 || set_values[2] > 31
+      errors.add(:date_of_birth, "not a valid day")
+      return
+    end
+      
+      self.send("#{multiparameter_attribute}=", Date.new(*set_values))
+    end
+  end
+
+  protected
+
+  def extract_multiparameter_attributes(new_attributes)
```

Comment:
What's this doing? It looks like a very complex way of assigning some attributes from a form?

Reply:
Exactly. The GovFormBuilder sets up 3 attributes for a date form, these are automagically combined when using ActiveRecord, but not with ActiveModel. So I have had to copy the AR method into the model. Its been a bit of a headscratcher TBH.

Reply:
The alternative is to create a custom form with a day, month, year attribute, which seems straightforwards, but it is not clear to me how to create a custom form with error fields using the gov style guide.

Reply:
see dob_spec.rb for how I have to build a test model